pkgs/development/python-modules/pyscf/default.nix

```diff
@@ -0,0 +1,67 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, cmake
+, openblas
+, libcint
+, libxc
+, h5py
+, numpy
+, scipy
+, pytest
+}:
+
+buildPythonPackage rec {
+  pname = "pyscf";
+  version = "1.7.0";
+
+  # must download from GitHub to get the Cmake & C source files
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0gdmc04nr699bq6gkw9isfzahj0k2gqhxnjg6gj9rybmglarkn15";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [
+    libcint
+    libxc
+    openblas
+  ];
+  cmakeFlags = [
+    # disable rebuilding/downloading the required libraries
+    "-DBUILD_LIBCINT=0"
+    "-DBUILD_LIBXC=0"
+    "-DBUILD_XCFUN=0"
+  ];
+  dontUseCmakeConfigure = true;
+
+  propagatedBuildInputs = [
+    h5py
+    numpy
+    scipy
+  ];
+
+  PYSCF_INC_DIR = lib.string.makeLibraryPath [ libcint libxc ];
+
+  pythonImportsCheck = [ "pyscf" ];
+  checkInputs = [ pytest ];
+  checkPhase = ''
+    pytest ./pyscf/tools/test
```

Comment:
Is that the official way to run the tests? Doesn't a plain `pytest` suffice?

Reply:
Sort of? I just looked into this, and it appears to have a lot of individual unittests, but my pytest always hangs on collecting the tests for some reason. Not able to easily trace where it's hanging.

It also has an extensive suite of examples, but most seem to be unguarded scripts (no if __name__ == "__main__"), and not in a standard test-runner format.

Will update when I find out more.

Reply:
I ran the tests in Travis CI build. I hadn't realized tests were distributed through the source dirs.
https://github.com/pyscf/pyscf/blob/master/.travis.yml